hazelcast/src/main/java/com/hazelcast/sql/impl/plan/Plan.java

```diff
@@ -77,7 +81,8 @@ public Plan(
         SqlRowMetadata rowMetadata,
         QueryParameterMetadata parameterMetadata,
         PlanCacheKey planKey,
-        Set<PlanObjectKey> objectIds
+        Set<PlanObjectKey> objectIds,
+        List<String> mapNames
```

Comment:
Have you considered using a builder pattern?
I know it's an "impl" class, but it would make the initializations more readable.

Reply:
I would say that both approaches (direct ctor and builder) have their own problems. With the builder, there is more code, and more ways to construct an inconsistent state. So I would prefer to leave it as is.